dev-java/protobuf-java/protobuf-java-3.20.1.ebuild

```diff
@@ -0,0 +1,90 @@
+# Copyright 2008-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# Tests not enabled, depend on com.google.truth which is not packaged
+# https://github.com/protocolbuffers/protobuf/blob/v21.4/java/core/pom.xml#L35-L40
+JAVA_PKG_IUSE="doc source"
+MAVEN_ID="com.google.protobuf:${PN}:${PV}"
+
+inherit java-pkg-2 java-pkg-simple
+
+PARENT_PN="${PN/-java/}"
+PARENT_PV="${PV}"
+PARENT_P="${PARENT_PN}-${PARENT_PV}"
+
+if [[ "${PV}" == "9999" ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/${PARENT_PN}"
+	EGIT_SUBMODULES=()
+	EGIT_CHECKOUT_DIR="${WORKDIR}/${PARENT_P}"
+else
+	inherit verify-sig
+	SRC_URI="
+		https://github.com/protocolbuffers/${PARENT_PN}/archive/v${PARENT_PV}.tar.gz
+			-> ${PARENT_P}.tar.gz
+	"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+fi
+
+DESCRIPTION="Google's Protocol Buffers - Java bindings"
+HOMEPAGE="
+	https://developers.google.com/protocol-buffers/
+	https://github.com/protocolbuffers/protobuf
+"
+
+LICENSE="BSD"
+SLOT="0/31"
+
+S="${WORKDIR}/${PARENT_P}/java"
+
+BDEPEND="
+	dev-libs/${PARENT_PN}:${SLOT}
+	>=virtual/jdk-1.8:*
```

Comment:
Drop jdk from BDEPEND, it's unneeded. See https://wiki.gentoo.org/wiki/Java_Developer_Guide#Dependencies

Reply:
Many thanks for this link, it will truly help me to make some clear choice now.
I will fix it very soon (and done for the ~~day~~ night) (and also for the rebase I forgot).